### PR TITLE
Format dynamics code

### PR DESCRIPTION
**Pull request by TomMelt**
# Format Dynamics Code

This PR brings the dynamics folder inline with the rest of the nextsimdg code.

- [x]  rename `dynamics/tests` folder to  `dynamics/test`
- [x] `clang-format` dynamics source 

**NOTE:** These changes are formatting only. There are no changes to the actual logic of the source code  

This will serve as the base for #556 which reinstates the dynamics tests

### CODE REVIEW

File: dynamics/src/DGTransport.cpp [hunk 1]
```diff
@@ -10,170 +10,152 @@
...
 {
-  return Eigen::Matrix<Nextsim::FloatType, 1, 3>(cv(eid, 0) + 0.5 * cv(eid, 2) + 1. / 6. * cv(eid, 4),
-						 cv(eid, 1) + 0.5 * cv(eid, 5) + 1. / 6. * cv(eid, 7),
-						 cv(eid, 3) + 0.5 * cv(eid, 6));
+    return Eigen::Matrix<Nextsim::FloatType, 1, 3>(
+        cv(eid, 0) + 0.5 * cv(eid, 2) + 1. / 6. * cv(eid, 4),
+        cv(eid, 1) + 0.5 * cv(eid, 5) + 1. / 6. * cv(eid, 7), cv(eid, 3) + 0.5 * cv(eid, 6));
 }
 
 // %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Comment (timspainNERSC) [inline dynamics/src/DGTransport.cpp:144]: If we are fixing the format, do we want to get rid of these section breaks? They aren't used anywhere else in the model.

File: dynamics/src/DGTransport.cpp [hunk 2]
```diff
@@ -218,44 +206,49 @@ void DGTransport<DG>::reinitnormalvelocity()
...
+                normalvel_Y.row((smesh.nx + 1) * iy + ix + 1) *= 2.0;
+            else if (seg == 2) // top
+                normalvel_X.row(smesh.nx * (iy + 1) + ix) *= 2.0;
+            else if (seg == 3) // left
+                normalvel_Y.row((smesh.nx + 1) * iy + ix) *= 2.0;
+        }
+    }
 }
 
 ////////////////////////////////////////////////// PREPARE
```
Comment (timspainNERSC) [inline dynamics/src/DGTransport.cpp:254]: Perhaps edit these back to 
```
// PREPARE
```
or
```
/********** PREPARE **********/
```
These are more common in C/C++ code

File: dynamics/src/DGTransport.cpp [hunk 3]
```diff
@@ -376,35 +388,44 @@ inline void DGTransport<1>::edge_term_Y(const ParametricMesh& smesh, const doubl
...
 
-    const LocalEdgeVector<EDGEDOFS(DG)> tmp =
-      (vel_gauss.array().max(0) * (rightedgeofcell<DG>(phi, c1) * PSIe<EDGEDOFS(DG), EDGEDOFS(DG)>).array() +
-       vel_gauss.array().min(0) * (leftedgeofcell <DG>(phi, c2) * PSIe<EDGEDOFS(DG), EDGEDOFS(DG)>).array());
+    const LocalEdgeVector<EDGEDOFS(DG)> tmp = (vel_gauss.array().max(0)
+            * (rightedgeofcell<DG>(phi, c1) * PSIe<EDGEDOFS(DG), EDGEDOFS(DG)>).array()
+        + vel_gauss.array().min(0)
+            * (leftedgeofcell<DG>(phi, c2) * PSIe<EDGEDOFS(DG), EDGEDOFS(DG)>).array());
 
     // - [[psi]] sind we're on the left side
```
Comment (timspainNERSC) [inline dynamics/src/DGTransport.cpp:430]: Spelling corrections in comments? (here: 'since')

File: dynamics/src/Interpolations.cpp [hunk 4]
```diff
@@ -26,7 +26,6 @@ namespace Interpolations {
 
     // ******************** Function -> DG ******************** //
```
Comment (timspainNERSC) [inline dynamics/src/Interpolations.cpp:27]: This is close enough to standard to be left as is.

File: dynamics/src/cgParametricMomentum.cpp [hunk 5]
```diff
@@ -34,166 +33,161 @@ namespace Nextsim {
...
+                                - v(i0_rt + CG + j * (CG * smesh.nx + 1))
+                                  << std::endl;
+                }
+            } else
+                abort();
+        }
+    }
+}
 
 // --------------------------------------------------
```
Comment (timspainNERSC) [inline dynamics/src/cgParametricMomentum.cpp:185]: Oh, will you just choose a section break format!

File: dynamics/src/cgParametricMomentum.cpp [hunk 6]
```diff
@@ -410,195 +384,159 @@ namespace Nextsim {
...
+    double dt_adv, const DGVector<3>& H, const DGVector<3>& A, DGVector<3>& D);
+template void CGParametricMomentum<1>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<6>& H, const DGVector<6>& A, DGVector<6>& D);
+
+template void CGParametricMomentum<2>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<1>& H, const DGVector<1>& A, DGVector<1>& D);
+template void CGParametricMomentum<2>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<3>& H, const DGVector<3>& A, DGVector<3>& D);
+template void CGParametricMomentum<2>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<6>& H, const DGVector<6>& A, DGVector<6>& D);
```
Comment (timspainNERSC) [inline dynamics/src/cgParametricMomentum.cpp:524]: I'm not sure I don't like the original format better. It's rather easier to read than the more compact `clang-format`ted version.

**Review by timspainNERSC — APPROVED** [review]
Approved, with comments about possible extra changes.

**Comment by TomMelt** [comment]
(copying from #556 )

> my plan is not to make any additional changes to [this PR], for the main reason I want it to just be changes relating to clang format and the renaming of the tests folder. Happy to make a separate PR to deal with the changes you suggested, but I don't want to add any additional confusion to an already tricky PR :sweat_smile:

**Comment by timspainNERSC** [comment]
> (copying from #556 )
> 
> > my plan is not to make any additional changes to [this PR], for the main reason I want it to just be changes relating to clang format and the renaming of the tests folder. Happy to make a separate PR to deal with the changes you suggested, but I don't want to add any additional confusion to an already tricky PR 😅

👍
Completely understandable!

**Review by einola — APPROVED** [review]

**Pull request merged**